src/core/manager.c

```diff
@@ -2284,6 +2284,32 @@ static int manager_dispatch_cgroups_agent_fd(sd_event_source *source, int fd, ui
         return 0;
 }
 
+static bool manager_process_barrier_fd(const char *buf, FDSet *fds) {
+        assert(buf);
+        assert(fds);
+
+        _cleanup_strv_free_ char **tags = NULL;
+
+        tags = strv_split(buf, NEWLINE);
```

Comment:
hmm, i#d prefer if we wouldn't have to allocate memory for this check, i.e. maybe rewrite as:

```c
if (STR_IN_SET(buf, "BARRIER=1", "BARRIER=1\n")) {
        …
        /* the two valid msg formats */
        …
} else if (startswith(buf, "BARRIER=1\n") || strstr(buf, "\nBARRIER=1\n" || endswith(buf, "\nBARRIER=1"))
        …
        /* invalid formats */
        …
}
```

you get the idea